### PR TITLE
#59 Add API to fetch feed posts

### PR DESCRIPTION
**Pull request by nike0503**
I am not sure if it is correct, please check

### CODE REVIEW

File: posts/migrations/0004_auto_20200503_1557.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    dependencies = [
+        ('posts', '0003_post_stream'),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
+            name='post',
+            options={'ordering': ['-last_modified']},
+        ),
```
Comment (aashishpiitk) [inline posts/migrations/0004_auto_20200503_1557.py]: remove migration files

File: README.md [hunk 2]
```diff
@@ -16,6 +16,12 @@ url : /users/unfollow/
 method : DELETE
 parameters = {"username" : "<username of the user to be unfollowed">}
 ```
+#### Fetch Feed Posts
+To see posts from followed users and streams
+```
+url : /users/feed/
```
Comment (aashishpiitk) [inline README.md:23]: shouldn't it be users/"<int:pk>"/feed
each user will have unique feed right

**Review by aashishpiitk — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (nike0503) [reply README.md:23]: But each user will see his own feed,  so once he is logged in there is no need for a primary key

File: posts/migrations/0004_auto_20200503_1557.py (hunk 1 above)
Comment (nike0503) [reply posts/migrations/0004_auto_20200503_1557.py]: Done

File: posts/models.py [hunk 3]
```diff
@@ -9,5 +9,8 @@ class Post(models.Model):
 	author = models.ForeignKey('users.User', on_delete= models.CASCADE)
 	stream = models.ForeignKey('streams.Stream', default=None, on_delete=models.CASCADE, db_column="title")
 
+	class Meta:
+		ordering = ['-pub_date']
```
Comment (namangup) [inline posts/models.py:13]: Wouldn't it be better to order by last modified ( descending ) ?

File: users/views.py [hunk 4]
```diff
@@ -113,6 +115,24 @@ def delete(self, request):
...
+            feed_posts_ids = list()
+            posts = Post.objects.all()
+            for post in posts:
+                if user in post.stream.followed_by.all():
+                    feed_posts_ids.append(post.id)
+                elif post.author in user.following.all():
+                    feed_posts_ids.append(post.id)
+            feed_posts = Post.objects.filter(id__in=feed_posts_ids)
+            serializer = PostSerializer(feed_posts, many=True)
+            return Response(serializer.data)
```
Comment (namangup) [inline users/views.py:133]: Use a try block maybe in case of any exceptions ?

File: posts/models.py (hunk 3 above)
Comment (nike0503) [reply posts/models.py:13]: I asked Divyanshu about this, he said it would be better to use publishing date

File: users/views.py (hunk 4 above)
Comment (d1vyanshu) [reply users/views.py:133]: objects.all() does not throw error in case of empty database, objects.get() does

**Pull request merged**